### ISSUE TITLE
Staggered grid

### ISSUE THREAD

**Issue by navidcy** [dhruvbhagtani2105/toy-model#8]
Just a placeholder to remind us that the need for a staggered grid is around the corner. As soon as we code the 1D shallow water equations we may start seeing some specific type of numerical instabilities which the staggered grid will heal.

**Comment by navidcy** [comment]
A nice discussion for why staggered grids are used can be found in [Oceananigans.jl Docs](https://clima.github.io/OceananigansDocumentation/dev/appendix/staggered_grid/). (The discussion is not particular to any programming language.)

Note that there is a bug and the schematic is not show. A [PR](https://github.com/CliMA/Oceananigans.jl/pull/1616) is submitted but until it's merged, have a look at the schematic [here](https://clima.github.io/OceananigansDocumentation/dev/numerical_implementation/finite_volume/).

**Comment by dhruvbhagtani** [comment]
Thanks for this documentation. I've been meaning to go through some paper/documentation as to why and how staggered grids are better. Seems like a good place to start.

**Comment by navidcy** [comment]
The easiest example to consider is the following:

centered difference scheme:

Everything is at rest. Only the fluid surface is non constant with variation 1m up and down every other grid point. 

What do you expect to happen physically?


What acceleration will the u momentum equation feel?

**Comment by dhruvbhagtani** [comment]
I'm not sure, but it shouldn't feel any acceleration right? Every alternate point will have the same height, so numerically, no spatial variations. But even in staggered grids, the height variable will be in the same location at all grids, so how does it help?

**Comment by navidcy** [comment]
> I'm not sure, but it shouldn't feel any acceleration right? Every alternate point will have the same height, so numerically, no spatial variations.

Yes you are right. The u-momentum equation won't feel any acceleration. But physically, if you have a zig-zag free surface as initial condition then lots of gravity waves will be emitted in either direction! So the numerics here will be saying that flow will remain at rest for ever and ever, very different from what the actual physics would do.

> But even in staggered grids, the height variable will be in the same location at all grids, so how does it help?

Yes, but the height variable **won't** be collocated with `u`. Therefore, every `u`-point will feel the gradient as computed from the two `h` points adjacent to it. Therefore if you have an `h` that varies up and down *every `h` point` the `u` **will** feel some acceleration. 

(if my explanation is not as clear, try making a small drawing and it will become clear I feel...)

**Comment by dhruvbhagtani** [comment]
Yes Navid, that makes sense. Drawing a figure removed all of my doubts.

**Comment by navidcy** [comment]
for the record,

<img src="https://user-images.githubusercontent.com/7112768/116757832-35b49780-aa52-11eb-9fa4-bb54f715bf9b.png" width = 600>
